godot/src/ui/components/label_avatar_name/label_avatar_name.gd

```diff
@@ -0,0 +1,99 @@
+@tool
+extends Control
+
+@export var avatar_name: String = "Kiko":
+	set(new_value):
+		avatar_name = new_value
+		_update_node()
+
+@export var avatar_subaddress: String = "1234":
+	set(new_value):
+		avatar_subaddress = new_value
+		_update_node()
+
+@export var avatar_has_claimed_name: bool = false:
+	set(new_value):
+		avatar_has_claimed_name = new_value
+		_update_node()
+
+@export var fit_text_to_label: bool = false:
+	set(new_value):
+		fit_text_to_label = new_value
+		_update_node()
+
+@export var max_font_size: int = 16:
+	set(new_value):
+		max_font_size = new_value
+		_update_node()
+
+@export var font: Font = null:
+	set(new_value):
+		font = new_value
+		_update_node()
+
+@export var hbox_alignament: BoxContainer.AlignmentMode = BoxContainer.ALIGNMENT_BEGIN:
+	set(new_value):
+		hbox_alignament = new_value
+		_update_node()
+
+
+func _update_node():
+	if not is_node_ready():
+		return
+
+	%Label_Name.text = avatar_name
```

Comment:
can we avoid using direct get_node instead of caching? I know it's a tool node but is it necessary as well?